internal/plugin/process_monitor_plugin.go

```diff
@@ -105,22 +105,45 @@ func (pm *ProcessMonitor) run(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case <-pm.processTicker.C:
-			newCtx := context.WithValue(ctx, logger.CorrelationIDContextKey, logger.GenerateCorrelationID())
-			slog.DebugContext(newCtx, "Checking for process changes")
+			slog.Debug("Checking for process updates")
```

Comment:
Why the change to just `slog.Debug()`? Maybe we should clarify when we should log with context and when not to 🤔

Reply:
my bad. It should be DebugContext. We should always pass context to the logger if it is available in the function.